[WIP] Support for stacking dataset

j-varun opened the pull request.

j-varun committed:
[WIP] Changes to add support for stacking dataset in micro search

j-varun committed:
[WIP] Bug Fixes

j-varun committed:
Minor Fixes

ahundt committed:
Merge pull request #3 from j-varun/good_fixes

[WIP] Support for stacking dataset

j-varun committed:
Added missing files

ahundt committed:
Merge pull request #3 from j-varun/good_fixes

[WIP] Support for stacking dataset

ahundt committed:
block_stacking_reader.py better reading test loop

ahundt committed:
add costar stacking search script

j-varun committed:
[WIP] Fixes for generator errors and other modifications to support stacking dataset

ahundt committed:
Merge commit 'd4757ac5df3e1d4a7e100dfcbabc9101d3d6372b' into stacking_search

* commit 'd4757ac5df3e1d4a7e100dfcbabc9101d3d6372b':
  [WIP] Fixes for generator errors and other modifications to support stacking dataset

# Conflicts:
#	enas/cifar10/block_stacking_reader.py

ahundt committed:
Merge branch 'stacking_search' of github.com:ahundt/enas into stacking_search

ahundt committed:
val_test_size to 200 for stacking

ahundt committed:
block stacking search batch size and num epochs updated

ahundt committed:
Merge commit 'b704f400aa11c580b5a6d432a55840870c898d0e' into stacking_search

* commit 'b704f400aa11c580b5a6d432a55840870c898d0e':
  fashion mnist final architecture with test accuracy 99.23%

j-varun committed:
[WIP]Changed Loss and Accuracy fn for regression task

j-varun committed:
[WIP] Added more metrics and arguments + minor bug fixes

j-varun committed:
Minor Fixes

j-varun committed:
[WIP] Minor changes to micro_child

ahundt committed:
Merge commit 'b774b87beae87c0af469d234e4ac4041f403185f' into stacking_search

* commit 'b774b87beae87c0af469d234e4ac4041f403185f':
  [WIP] Minor changes to micro_child
  Minor Fixes
  [WIP] Added more metrics and arguments + minor bug fixes
  [WIP]Changed Loss and Accuracy fn for regression task

# Conflicts:
#	enas/cifar10/models.py

ahundt committed:
models.py pep8

j-varun committed:
[WIP] Fixed hardcoded shapes

j-varun committed:
[WIP] Minor Bug Fixes

j-varun committed:
Merged with stacking_search

ahundt committed:
correct validation size

j-varun committed:
bug fix

ahundt committed:
Merge commit 'a5f63525eb0bba501dc4ab4d194f0239984857aa' into stacking_search

* commit 'a5f63525eb0bba501dc4ab4d194f0239984857aa':
  bug fix

# Conflicts:
#	enas/cifar10/models.py

ahundt committed:
models.py reduce images per example to train on for temp bug workaround.

ahundt committed:
models.py hack to reduce time in a batch

ahundt committed:
cifar10 main.py add option for default child and controller optimizers

j-varun committed:
Miscellaneous Bug Fixes

j-varun committed:
[WIP] Changes to support micro search based on loss and other fixes

j-varun committed:
[WIP] Changes for using validation loss as reward for controller

ahundt committed:
Merge commit '9e3dc292dd485d65bf053e587d4b486190c553ed' into stacking_search

* commit '9e3dc292dd485d65bf053e587d4b486190c553ed':
  [WIP] Changes to support micro search based on loss and other fixes
  Miscellaneous Bug Fixes

# Conflicts:
#	enas/cifar10/main.py
#	enas/cifar10/models.py

ahundt committed:
models.py indentation fix

ahundt committed:
models.py fix indentation

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

j-varun committed:
[WIP] Minor Fix for initializing child class

j-varun committed:
[WIP] Removed unnecessary print statements

ahundt committed:
increase training # of workers and max queue size

ahundt committed:
costar_block_stacking_search.sh attempt at tuning parameters

ahundt committed:
costar_block_stacking_search.sh attempt at tuning parameters

ahundt committed:
global_avg_pool -> global_max_pool

ahundt committed:
costar_block_stacking_search.sh reduce batch size due to memory constraints

ahundt committed:
lower child out filters to fit in memory

j-varun committed:
[WIP] Fixes for fashion_mnist and stacking dataset

j-varun committed:
Minor bug fix

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

ahundt committed:
models.py artifically lower number of steps in an epoch

ahundt committed:
micro_child.py lambda to def

ahundt committed:
models.py make stacking epoch longer by batch_size

j-varun committed:
Added additional evaluation metrics

j-varun committed:
[WIP] Minor bug fix

j-varun committed:
[WIP] Minor fixes

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

j-varun committed:
option for translation only training added

j-varun committed:
[WIP] Minor Fixes

ahundt commented on enas/cifar10/block_stacking_reader.py:
```diff
@@ -479,7 +479,7 @@ def ConvertImageListToNumpy(data, format='numpy', data_format='NHWC'):
                 raise ValueError('Unsupported data input: ' + str(self.data_features_to_extract))
 
-            if (self.data_features_to_extract is not None and 'image_0_image_n_vec_xyz_aaxyz_nsc_15' in self.data_features_to_extract):
+            if (self.data_features_to_extract is not None and x in self.data_features_to_extract for x in ['image_0_image_n_vec_xyz_aaxyz_nsc_15', 'image_0_image_n_vec_xyz_10']):
```
this line change seems strange maybe use an or statement? I'm concerned in some cases this won't evaluate the way we would like, since I believe it will return a list.

j-varun committed:
Fixes for translation only run

j-varun committed:
Added Metrics for eval

j-varun committed:
Bug Fix

j-varun committed:
Fix for fashion-mnist

j-varun committed:
Added option to switch to alternate reward for stacking dataset

j-varun committed:
Added flad for reward

j-varun committed:
Changes to support training of only the rotation component of the stacking dataset

j-varun committed:
Minor Fixes

ahundt committed:
Merge pull request #5 from j-varun/good_fixes

Stacking search, support separate and joint translation / rotation

j-varun committed:
Minor Bug Fix

ahundt committed:
configure search parameters for rotation and translation only search

ahundt committed:
Merge commit 'f0f440a58a225f17c6b2457f1d0b0d763c47e802' into stacking_search

* commit 'f0f440a58a225f17c6b2457f1d0b0d763c47e802':
  Minor Bug Fix

j-varun committed:
Added Configurable max loss for reward

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

j-varun committed:
Miscellaneous Fixes

ahundt committed:
Merge pull request #6 from j-varun/good_fixes

clearer printed labels/values

ahundt committed:
costar_block_stacking_translation_search.sh last training run only found extreme translations, try new training configuration

ahundt committed:
models.py increase estimated images per example from 1 to 15

ahundt committed:
models.py estimated images per example = 16

ahundt committed:
costar_block_stacking_rotation_search.sh configure to match translation search

ahundt committed:
rotation search try increasing batch size to 32

ahundt committed:
configure new search directories

ahundt committed:
rotation search batch size 16

ahundt committed:
costar stacking lower estimated images per example to 4

ahundt committed:
models.py lower estimated_images_per_example to 2

j-varun committed:
Fixes for accuracy values being printed

ahundt committed:
batch_norm -> group_norm with exception of masked batch norm...

ahundt committed:
set max_loss to 2 to avoid negative losses

ahundt committed:
Merge commit '6e232e573515826ce172a3d0bb45154193c409ff' into stacking_search

* commit '6e232e573515826ce172a3d0bb45154193c409ff':
  Fixes for accuracy values being printed

ahundt committed:
image_ops.py attempt group_norm fixes

ahundt committed:
group_norm try to fix shape problem

ahundt committed:
image_ops.py group_norm attempt to fix shape

ahundt committed:
image_ops.py group_norm attempt at shape fix

ahundt committed:
image_ops.py more attempts at group norm fix

ahundt committed:
image_ops.py group_norm fix attempt

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
[1, C, 1, 1]

ahundt committed:
image_ops.py debug shape

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
iimage_ops.py group_norm another shape fix attempt

ahundt committed:
micro_child.py fixes to is_training parameters

ahundt committed:
image_ops.py get some additional debug info

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
image_ops.py norm prints traceback if verbose

ahundt committed:
typo fix

ahundt committed:
image_ops.py group_norm another shape fix attempt

ahundt committed:
micro_child.py add shape asserts

ahundt committed:
micro_child.py more explicit params

ahundt committed:
image_ops.py print clearer separator

ahundt committed:
micro_child.py clearly set is training as named parameter

ahundt committed:
micro_child.py better setting of is_training

ahundt committed:
mico_child.py typo fix

ahundt committed:
micro_child.py fix some hard coded variables

ahundt committed:
micro_child.py typo fix

ahundt committed:
micro_child.py typo fix

ahundt committed:
micro_child.py add missing import

ahundt committed:
micro_child.py add more shape debugging

ahundt committed:
micro_chilld.py more shape debugging

ahundt committed:
micro_child.py more shape debug output

ahundt committed:
micro_child.py more shape debug output

ahundt committed:
micro_child.py more shape debug outputs

ahundt committed:
micro_child.py try using integer shape components where possible

ahundt committed:
micro_child.py typo fix

ahundt committed:
image_ops.py micro_child.py set verbose shape debug output to 0

ahundt committed:
models.py CRITICAL BUGFIX: TRAIN GENERATOR WAS BEING USED FOR VAL + TEST!!!

ahundt committed:
micro_child.py fix is_training parameters

ahundt committed:
cifar10 parameterize pool distance

ahundt committed:
double num_layers so receptive field size is increased substantially

ahundt committed:
main.py set pool distance based on flag

ahundt committed:
block_stacking_reader.py merge code from costar_plan

ahundt committed:
block_stacking_reader.py incorporate upstream changes and add new input configuration

ahundt committed:
micro search add missing flags and an error message.

ahundt committed:
models.py translation_only add xy grid and remove input rotations

ahundt committed:
micro_child.py change aux head training config

ahundt committed:
costar_block_stacking_translation_search.sh 64 filters

ahundt committed:
general_child.py, micro_child.py ALL RELU -> ELU

ahundt committed:
micro_child.py typo fix

ahundt committed:
block_stacking_reader.py fix major bugs in data loading code for data cubes

ahundt committed:
block_stacking_reader.py fix assert bug

ahundt committed:
block_stacking_reader.py disable debug code

ahundt committed:
rotation_search.sh batch_size=16

j-varun committed:
Changed output for test and validation to include arc

j-varun committed:
Added new case with reward estimates in block_stacking_reader.py

ahundt committed:
Merge commit '83b74cd0530fa2ce3cc58f98aae25b809944e965' into stacking_search

* commit '83b74cd0530fa2ce3cc58f98aae25b809944e965':
  Changed output for test and validation to include arc

j-varun committed:
Added support for reward_estimates training case

ahundt committed:
main.py typo fix

ahundt committed:
set batch size and filters to work on titan xp

ahundt committed:
add requirements.txt

ahundt committed:
Merge commit '676f082dbdf7d5055a74df996e097f53a944d458' into stacking_search

* commit '676f082dbdf7d5055a74df996e097f53a944d458':
  Added support for reward_estimates training case
  Added new case with reward estimates in block_stacking_reader.py

ahundt committed:
Merge commit 'bf8506b94c7eeda1f42d5d0ba2888c13b036c3ac' into stacking_search

* commit 'bf8506b94c7eeda1f42d5d0ba2888c13b036c3ac':
  add requirements.txt
  set batch size and filters to work on titan xp
  main.py typo fix

ahundt committed:
Add reward critic search script

ahundt committed:
set reward critic output directory

ahundt committed:
reward_estimate -> stacking_reward so it can be told apart from the enas controller reward

ahundt committed:
TRANSLATION SEARCH MAJOR REWARD CHANGE TO 1/MSE

ahundt committed:
block_stacking_reader.py only check for length of goal ids if it is less than the number of goal ids

ahundt committed:
rename stacking reward variable

ahundt committed:
rotation search reduce batch size for GTX 1080 Ti

ahundt committed:
micro rotation search partial reversion of settings (filters & input layers) to run from:
http://femur.jhu.edu:4005/2018_08_12_2054_micro_rotation_search_output_stack_DECENT_RESULTS.txt

ahundt committed:
attempt 2 at printing eval architecture

ahundt committed:
micro_child.py convert arc data to tensor so it will work in fixed and non-fixed case

ahundt committed:
build_valid() is_training = True

ahundt committed:
arc print fix

ahundt committed:
micro_child.py disable printing arcs continuously, but make it easy to re-enable

ahundt committed:
add and enable random augmentation for translation search only

ahundt committed:
typo fix

ahundt committed:
main.py add random_augmentation parameter to child class call

ahundt committed:
create rotation final script

ahundt committed:
rotation final fix nchw -> nhwc

ahundt committed:
rotation search and final command fixes

j-varun committed:
Temporary fix for group_norm issues

j-varun committed:
Another temporary fix for norm issues

j-varun committed:
Switch to batch_norm until group_norm is fixed

j-varun committed:
Changes to hyperparameters for rotation

j-varun committed:
Fix Typo

j-varun committed:
Changes in rotation search script

j-varun committed:
Changes for costar_block_stacking_v0.3, load files from train-test-val text files, additional metrics and other minor fixes

j-varun committed:
Metrics being saved to csv

j-varun committed:
Changes in script for costar_block_stacking_v0.3

j-varun committed:
updated names of text files

ahundt committed:
block_stacking_reader.py expand user path to filename

j-varun committed:
Fix in csv file write mode for Python2 and minor changes in scripts

j-varun committed:
Minor bug fix

j-varun committed:
fix for csv file handling

ahundt committed:
micro_child.py attempt to add printing of arc during eval and test

j-varun committed:
Bug fixes

ahundt committed:
Merge commit '91cd02b5660ad3d85602faa685cc89084ac1faaf' into stacking_search

* commit '91cd02b5660ad3d85602faa685cc89084ac1faaf':
  fix for csv file handling
  Minor bug fix
  Fix in csv file write mode for Python2 and minor changes in scripts

ahundt committed:
Merge commit '661db8e89ee9609f04b4001b6588c44c11939161' into stacking_search

* commit '661db8e89ee9609f04b4001b6588c44c11939161':
  Bug fixes

# Conflicts:
#	enas/cifar10/micro_child.py

ahundt committed:
micro_child.py print_arc handle feed dict None case

j-varun committed:
Additional metrics + Arc print for eval

j-varun committed:
Changed rotation_weight to 1 for encoding and decoding in grasp_metrics, max pooling in reduction and updated training scripts

ahundt committed:
Merge commit '6b7631796bc5d3c5c96205af1374f924e90f722a' into stacking_search

* commit '6b7631796bc5d3c5c96205af1374f924e90f722a':
  Changed rotation_weight to 1 for encoding and decoding in grasp_metrics, max pooling in reduction and updated training scripts
  Additional metrics + Arc print for eval

j-varun committed:
Fixes for typos

ahundt committed:
Merge commit 'c97b14544ff6d316fbaac3071ee16eba6a58616f' into stacking_search

* commit 'c97b14544ff6d316fbaac3071ee16eba6a58616f':
  Fixes for typos

j-varun committed:
More typo fixes

j-varun committed:
Commenting unsuccessful printing of arcs

ahundt committed:
Merge commit 'e2336be9d9320dcfbba9fbb67671bf31f1a1edfa' into stacking_search

* commit 'e2336be9d9320dcfbba9fbb67671bf31f1a1edfa':
  Commenting unsuccessful printing of arcs
  More typo fixes

# Conflicts:
#	enas/cifar10/micro_child.py

j-varun committed:
Added support for inference in reader

ahundt committed:
Merge commit 'c46e27d346185ec98f603a5d4176f15182929545' into stacking_search

* commit 'c46e27d346185ec98f603a5d4176f15182929545':
  Added support for inference in reader

j-varun committed:
Bug fixes for metrics

j-varun committed:
Bug fixes for metrics

j-varun committed:
inference mode function integrated into generator in block_stacking_reader.py

j-varun committed:
Option to switch to one hot encoding in block_stacking_reader.py and added convolution and dense layer root to ENAS

j-varun committed:
Added ELU activation to convolutions in new root

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

ahundt committed:
Merge commit 'b19806d129e9dd0a696d9fcd96a3e70c4e070a52' into stacking_search

* commit 'b19806d129e9dd0a696d9fcd96a3e70c4e070a52':
  Added ELU activation to convolutions in new root
  Option to switch to one hot encoding in block_stacking_reader.py and added convolution and dense layer root to ENAS
  inference mode function integrated into generator in block_stacking_reader.py
  Bug fixes for metrics
  Bug fixes for metrics

j-varun committed:
Commented print_arc in eval

ahundt committed:
Merge commit '5eb698b676b435d98112fdb80027e46fe6d1b581' into stacking_search

* commit '5eb698b676b435d98112fdb80027e46fe6d1b581':
  Commented print_arc in eval

j-varun committed:
Bug fix and updated scripts"

ahundt committed:
Merge commit '0cb8aa65e2676418f90a7617bc3ae3bde93edc8e' into stacking_search

* commit '0cb8aa65e2676418f90a7617bc3ae3bde93edc8e':
  Bug fix and updated scripts"

ahundt committed:
merge changes from costar_plan

ahundt committed:
block_stacking_reader.py CRITICAL MERGE OF SPLIT OUT ENCODING API

j-varun committed:
Changes in stem_conv

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

ahundt committed:
set new rotation final run, make sure val and test is collected

ahundt committed:
Create block stacking translation final script

ahundt committed:
Merge pull request #8 from j-varun/stacking_search

translation final run

j-varun committed:
Option to use MSLE as primary loss function

j-varun committed:
Merge branch 'stacking_search' of https://github.com/ahundt/enas into good_fixes

j-varun committed:
Changes in rotation final run

j-varun committed:
Path fixes

j-varun committed:
Additional condition check in reader

j-varun committed:
Batch size change in rotation final script

ahundt committed:
Use model batch with low angular error and 100% acc

ahundt committed:
enas/cifar10/block_stacking_reader.py fix bad merge

j-varun committed:
Fix for redundant condition

ahundt committed:
enas/cifar10/block_stacking_reader.py fix anoter merge problem

ahundt committed:
Merge branch 'stacking_search' into good_fixes

ahundt committed:
Merge pull request #9 from j-varun/good_fixes

Merge update for final rotation run

ahundt committed:
Fix some details for starting the run

ahundt committed:
batch size 32

ahundt committed:
fix csv path

j-varun committed:
fix for running with msle

j-varun committed:
Typo fix

ahundt committed:
final rotation search settings, add no root search scripts

j-varun committed:
New scripts for final run and added headers for csv

j-varun committed:
Timestamp update in scripts
